packages/semi-foundation/progress/generates.ts

```diff
@@ -0,0 +1,354 @@
+    }
+    return undefined;
+}
+
+const toHex = {
+    ColorName(color: string): string {
+        return SET_NAME_HEX[color];
+    },
+    Hex(color: string, transparency: string | undefined): string {
+        color = color.replace('#', '');
+        if (color.length === 8) return `#${color}`;
+        if (color.length === 6) return `#${color}${transparency || 'ff'}`;
+        if (color.length === 3) {
+            color = color
+                .split('')
+                .map(c => c + c)
+                .join('');
+        }
+        return `#${color}${transparency || 'ff'}`;
+    },
+    HslA(color: string): string {
+        const hsla = REG_S.hslA.exec(color);
+        const h = parseInt(hsla[2]);
+        const s = parseInt(hsla[3]) / 100;
+        const l = parseInt(hsla[4]) / 100;
+        const a = hsla[5];
+        const c = (1 - Math.abs(2 * l - 1)) * s,
+            x = c * (1 - Math.abs(((h / 60) % 2) - 1)),
+            m = l - c / 2;
+        let r: string | number = 0,
+            g: string | number = 0,
+            b: string | number = 0;
+        if (0 <= h && h < 60) {
+            r = c;
+            g = x;
+            b = 0;
+        } else if (60 <= h && h < 120) {
+            r = x;
+            g = c;
+            b = 0;
+        } else if (120 <= h && h < 180) {
+            r = 0;
+            g = c;
+            b = x;
+        } else if (180 <= h && h < 240) {
+            r = 0;
+            g = x;
+            b = c;
+        } else if (240 <= h && h < 300) {
+            r = x;
+            g = 0;
+            b = c;
+        } else if (300 <= h && h < 360) {
+            r = c;
+            g = 0;
+            b = x;
+        }
+        r = Math.round((r + m) * 255).toString(16);
+        g = Math.round((g + m) * 255).toString(16);
+        b = Math.round((b + m) * 255).toString(16);
+        return toHex.utils.pAL(r, g, b, a);
+    },
+    RgbA(color: string): string {
+        const rgba = REG_S.rgbA.exec(color);
+        const r = parseInt(rgba[2], 10).toString(16),
+            g = parseInt(rgba[3], 10).toString(16),
+            b = parseInt(rgba[4], 10).toString(16),
+            a = rgba[5];
+        return toHex.utils.pAL(r, g, b, a);
+    },
+    utils: {
+        pAL(r: string, g: string, b: string, a: string) {
+            if (r.length == 1) r = '0' + r;
+            if (g.length == 1) g = '0' + g;
+            if (b.length == 1) b = '0' + b;
+            if (typeof a !== 'undefined') {
+                a = Math.round(parseInt(a) * 255).toString(16);
+                if (a.length == 1) a = '0' + a;
+                return '#' + r + g + b + a;
+            }
+            return '#' + r + g + b;
+        },
+    },
+};
+
+const REG_S = {
+    hex: /^#([0-9a-fA-F]{3}|[0-9a-fA-F]{6}|[0-9a-fA-F]{8})$/,
+    hslA: /(hsl)a?\((\d+),?\s*?(\d+)%,?\s*?(\d+)%,?\s*?\/?(\s*?[\d.]+)?\)/,
+    rgbA: /(rgb)a?\((\d+),?\s*?(\d+),?\s*?(\d+),?\s*?\/?(\s*?[\d.]+)?\)/,
+};
+
+const SET_NAME_HEX = {
```

Comment:
I have a little opinion of. SET_NAME_HEX setting
1. The color naming here is different from Semi's naming of color tokens.（https://semi.design/en-US/basic/tokens）  In a design system, if the color name of a single component does not correspond to the global color name. It will increase the user's memory cost. I suggest align with global token,  also i think provide 16 basic color is enough.

2. If a fixed hex color value is used here, it cannot adapt to Semi dark mode.  I think you can read color mapping from body element like (but there are rba format, you need to covert)
```
getComputedStyle(document.body).getPropertyValue('--semi-blue-5')
```

![image](https://user-images.githubusercontent.com/88709023/188832454-034b249e-8ca8-43ec-8a62-ed081f1ba3fb.png)

Reply:
Indeed, this does need to be amended 🤔

Reply:
I suggest that the colorname here is aligned with the Semi color palatte naming. Then it supports specific color levels with numbers, and also supports use without numbers (5 can be selected by default at this time)

eg
```
name : cssVariable

'blue' -> semi-blue-5
'blue-4' -> semi-blue-4
'green' -> semi-green-5
'green-1' -> semi-green-1
```

Reply:
Great! I was thinking about how to match the Semi color palette before.